opentelemetry-sdk/src/metrics/exporter.rs

```diff
@@ -10,17 +8,19 @@ use super::Temporality;
 /// Exporter handles the delivery of metric data to external receivers.
 ///
 /// This is the final component in the metric push pipeline.
-#[async_trait]
 pub trait PushMetricExporter: Send + Sync + 'static {
     /// Export serializes and transmits metric data to a receiver.
     ///
     /// All retry logic must be contained in this function. The SDK does not
     /// implement any retry logic. All errors returned by this function are
     /// considered unrecoverable and will be logged.
-    async fn export(&self, metrics: &mut ResourceMetrics) -> OTelSdkResult;
+    fn export(
+        &self,
+        metrics: &mut ResourceMetrics,
+    ) -> impl std::future::Future<Output = OTelSdkResult> + Send;
 
     /// Flushes any metric data held by an exporter.
-    async fn force_flush(&self) -> OTelSdkResult;
+    fn force_flush(&self) -> impl std::future::Future<Output = OTelSdkResult> + Send;
```

Comment:
just realized that this is something we need to remove, in a separate PR. Flush can be normal method, consistent with other signals.